src/metal.cs

```diff
@@ -405,11 +490,31 @@ public partial interface MTLDevice {
 		[NoTV]
 		[Export ("headless")]
 		bool Headless { [Bind ("isHeadless")] get; }
+		
+		[NoiOS, NoTV, NoWatch, Mac (10,12)]
+		[Abstract]
+		[Export ("recommendedMaxWorkingSetSize")]
+		ulong RecommendedMaxWorkingSetSize { get; }
 
 		[NoiOS]
 		[NoTV]
 		[Export ("depth24Stencil8PixelFormatSupported")]
 		bool Depth24Stencil8PixelFormatSupported { [Bind ("isDepth24Stencil8PixelFormatSupported")] get; }
+		
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
+		[Export ("heapTextureSizeAndAlignWithDescriptor:")]
+		MTLSizeAndAlign HeapTextureSizeAndAlign (MTLTextureDescriptor desc);
+
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
+		[Export ("heapBufferSizeAndAlignWithLength:options:")]
+		MTLSizeAndAlign HeapBufferSizeAndAlignWithLength (nuint length, MTLResourceOptions options);
+
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
```

Comment:
same - you can add a comment on them and you'll need to teach **xtro** about the special case too